### ISSUE TITLE
Crash encountered when caching scenes for reuse.

### ISSUE THREAD

**Issue by avencherus** [godotengine/godot#17793]
This is currently happening in 2.1.4 and 2.1.5 RC1 using Windows 10.  I couldn't reproduce it in 3.0.2, so it seems fine there.

The jist of the problem is that I'm instancing scenes that are RigidBody2D, and track their use and availability with a dictionary.  Once they are done, they will remove themselves from the scene tree.  This works fine up until this list is no longer being requested and the last active scene is removed from the tree.

Luckily for once I was able to recreate it in a demo project at least on my end.  Just spam the button a little and then wait for the last item to get recycled.

[Scene Caching Bug 2.1.4.zip](https://github.com/godotengine/godot/files/1851468/Scene.Caching.Bug.2.1.4.zip)

Seems like a minor thing to work around, but reporting it in case it is something that runs deeper somewhere else.

**Comment by akien-mga** [comment]
Backtrace (2.1.5-stable):
```
Thread 1 "godot-2.1" received signal SIGSEGV, Segmentation fault.
0x0000000001e13a8c in SelfList<Body2DSW>::List::remove (this=0x3c8a228, p_elem=0x0) at core/self_list.h:57
57                              ERR_FAIL_COND(p_elem->_root != this);
(gdb) bt
#0  0x0000000001e13a8c in SelfList<Body2DSW>::List::remove (this=0x3c8a228, p_elem=0x0) at core/self_list.h:57
#1  0x0000000001e125be in Space2DSW::call_queries (this=0x3c8a1d0) at servers/physics_2d/space_2d_sw.cpp:1149
#2  0x0000000001d1f97c in Physics2DServerSW::flush_queries (this=0x37524f0) at servers/physics_2d/physics_2d_server_sw.cpp:1218
#3  0x0000000001d02bbd in Physics2DServerWrapMT::flush_queries (this=0x3761100) at servers/physics_2d/physics_2d_server_wrap_mt.cpp:99
#4  0x0000000000c57946 in Main::iteration () at main/main.cpp:1522
#5  0x0000000000c38187 in OS_X11::run (this=0x7fffffffd4c0) at platform/x11/os_x11.cpp:2314
#6  0x0000000000c2e03c in main (argc=1, argv=0x7fffffffda58) at platform/x11/godot_x11.cpp:51
```

**Comment by akien-mga** [comment]
It seems related to #10886, which was fixed by 4deac219b.

**Comment by akien-mga** [comment]
Fixed by 531fb3b8.
